nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -155,8 +155,7 @@ in
 
     systemd.services.dhcpcd = let
       cfgN = config.networking;
-      hasDefaultGatewaySet = (cfgN.defaultGateway != null && cfgN.defaultGateway.address != "")
-                          || (cfgN.defaultGateway6 != null && cfgN.defaultGateway6.address != "");
+      hasDefaultGatewaySet = (cfgN.defaultGateway != null && cfgN.defaultGateway.address != "");
```

Comment:
That this is invalid too -- this would prevent starting dhcpcd for resolving IPv6 addresses before declaring a network online if IPv4 gateway is set. We probably want `network-online.target` to signify that both stacks are ready.

Reply:
@abbradar you are perfectly right. I'm closing this in favour of #35141

Reply:
Nope, you actually have n advantage of yours: you also fix `hasDefaultGatewaySet` in `network-interface-scripted.nix` which I didn't know was there :3 I think it's better to unify those as a hidden NixOS option...

Reply:
I'll do this in #35141.